src/cache.rs

```diff
@@ -62,7 +62,7 @@ pub struct Cache {
     /// Map containing for each FileIDs a list of files they import.
     imports: HashMap<FileId, HashSet<FileId>>,
     /// The table storing parsed terms corresponding to the entries of the file database.
-    terms: HashMap<FileId, (RichTerm, EntryState)>,
+    terms: HashMap<FileId, (RichTerm, EntryState, ParseErrors)>,
```

Comment:
Nitpicking: it might be worth making a separate struct `CacheEntry` or something, for readability and ease of pattern matching.

Reply:
Refactored